insitu/views/reportsmixins.py

```diff
@@ -0,0 +1,388 @@
+
+class ReportExcelMixin:
+
+    def set_formats(self, workbook):
+        self.merge_format = workbook.add_format({
+            'bold': 1,
+            'align': 'center',
+            'valign': 'vcenter',
+            'font_name': 'Calibri',
+            'font_size': 14,
+            'font_color': '#00B050',
+        })
+
+        self.format_header = workbook.add_format({
+            'bold': 1,
+            'align': 'left',
+            'valign': 'vcenter',
+            'font_name': 'Calibri',
+            'font_size': 14,
+            'font_color': 'red',
+        })
+
+        self.format_cols_headers = workbook.add_format({
+            'bold': 1,
+            'align': 'center',
+            'valign': 'vcenter',
+            'font_name': 'Calibri',
+            'font_size': 12,
+            'font_color': '#0070C0',
+            'bg_color': '#c3d69b',
+            'border': 1,
+        })
+
+        self.format_rows = workbook.add_format({
+            'align': 'left',
+            'valign': 'vcenter',
+            'font_name': 'Calibri',
+            'font_size': 12,
+            'text_wrap': 1,
+            'border': 1,
+        })
+
+    def generate_header_sheet(self, workbook, worksheet):
+        worksheet.set_column('A1:D1', 30)
+        worksheet.set_row(0, 30)
+        worksheet.set_row(2, 20)
+        worksheet.set_row(3, 20)
+        worksheet.set_row(5, 20)
+        worksheet.merge_range(
+            'A1:D1', 'Copernicus In Situ Component Information System - managed by the European Environment Agency', self.merge_format)
+        worksheet.merge_range(
+            'A3:D3', 'Standard Report for Local Land Component', self.merge_format)
+        worksheet.merge_range(
+            'A4:D4', 'Produced on 30th October 2020', self.merge_format)
+        worksheet.merge_range(
+            'A6:D6', 'The Standard Report consists of tables that include  all the main  statistical data  . . . . .')
+        worksheet.merge_range('A7:D7', 'The objects in this document are filtered using the following services: {} and the following components: {}'.format(
+            ", ".join([elem.name for elem in self.services]),
+            ", ".join([elem.name for elem in self.components])))
+
+    def generate_table_1(self, workbook, worksheet):
+        worksheet.set_column('A1:B1', 20)
+        worksheet.set_column('B1:C1', 50)
+        worksheet.set_column('D1:K1', 25)
+        worksheet.set_row(0, 30)
+        worksheet.set_row(1, 20)
+        worksheet.merge_range(
+            'A1:K1', 'REQUIREMENTS AND THEIR DETAILS', self.format_header)
+        headers = [
+            'REQUIREMENT UID', 'REQUIREMENT', 'NOTE', 'DISSEMINATION', 'QUALITY CONTROL', 'GROUP',
+            'UNCERTAINTY (%)', 'UPDATE FREQUENCY', 'TIMELINESS', 'SCALE', 'HORIZONTAL RESOLUTION'
+        ]
+        worksheet.write_row('A2', headers, self.format_cols_headers)
+        self.requirements = Requirement.objects.filter(
+            products__in=self.products).distinct()
```

Comment:
If you reeeally want to optimize here, you could use `values_list` to avoid loading a whole Requirement instance for each database entry.